File: rust/shape/tamil.rs
use super::{shared::shared_shaper, TAMIL_VOWELS};

/// Check if a unicode character is Tamil
pub fn is_tamil(c: &u16) -> bool {
    *c >= 0x0b80 && *c <= 0x0bff
}

/// Shape Tamil unicode
pub fn shape_tamil(input: &mut [u16]) {
    shared_shaper(input, &TAMIL_VOWELS);
}

File: src/shape/tamil.ts
import { TAMIL_VOWELS } from './internal.js';
import { sharedShaper } from './shared.js';

/**
 * Check if a character is Tamil
 * @param unicode - input unicode character
 * @returns - True if Tamil
 */
export function isTamil(unicode: number): boolean {
  return unicode >= 0x0b80 && unicode <= 0x0bff;
}

/**
 * Converts Tamil Unicode buffer into a shaped Tamil Unicode buffer
 * @param input - input unicode buffer
 */
export function shapeTamil(input: number[]): void {
  sharedShaper(input, TAMIL_VOWELS);
}
